### PR TITLE
feat: add demoUser role to Hasura with permissions

### PR DESCRIPTION
**Pull request by RODO94**
## What does this PR do? 

For the Demo Workspace, a `demoUser` may only see flows created by them, or flows within the Open Systems Lab, Open Digital Planning, or Templates team. Also, they cannot see team settings or any other setting outwith flow settings for their own flows.

I have treated them as a `teamEditor` within their flows, and restricted data they can see outwith this. 

### Introspection testing

Introspection testing has been added to help communicate the relevant permissions changes and ensure future maintainability.

### CODE REVIEW

File: hasura.planx.uk/metadata/tables.yaml [hunk 1]
```diff
@@ -252,6 +252,24 @@
       using:
         foreign_key_constraint_on: team_id
   insert_permissions:
+    - role: demoUser
```
Comment (DafyddLlyr) [inline hasura.planx.uk/metadata/tables.yaml:255]: I know this is still a draft, but just a heads up that this is already an action granted by the `public` role and doesn't need to be repeated for the `demo` role.

File: hasura.planx.uk/metadata/tables.yaml [hunk 2]
```diff
@@ -566,7 +647,7 @@
         computed_fields:
           - data_merged
         filter: {}
-        allow_aggregations: true
```
Comment (RODO94) [inline hasura.planx.uk/metadata/tables.yaml:569]: @DafyddLlyr see this as a change, but don't remember changing it. Any idea what it is and how to change it back?

Comment (DafyddLlyr) [reply hasura.planx.uk/metadata/tables.yaml:569]: Good spot.

It looks like this has been checked somewhere https://hasura.io/docs/2.0/auth/authorization/permissions/aggregation-permissions/

There aren't many places we use these types of query - I'd check what we do for another corresponding permission set.

File: hasura.planx.uk/metadata/tables.yaml [hunk 3]
```diff
@@ -530,6 +578,39 @@
...
+                    id:
+                      _eq: 32
+            - team:
+                _or:
+                  - id:
+                      _eq: 1
+                  - id:
+                      _eq: 29
+                  - id:
+                      _eq: 30
```
Comment (DafyddLlyr) [inline hasura.planx.uk/metadata/tables.yaml:612]: Not sure why I didn't spot this earlier, but maybe the `_in` operator would be a good alternative here?

File: hasura.planx.uk/tests/email_applications.test.js [hunk 4]
```diff
@@ -60,6 +60,20 @@ describe("email_applications", () => {
       expect(i).toHaveNoMutationsFor("email_applications");
     });
   });
+  describe("demoUser", () => {
```
Comment (DafyddLlyr) [inline hasura.planx.uk/tests/email_applications.test.js:63]:
```suggestion
  });
  
  describe("demoUser", () => {
```

File: hasura.planx.uk/tests/feedback_status.test.js [hunk 5]
```diff
@@ -1,92 +1,106 @@
...
+
+    test("cannot query feedback_status_enum", () => {
+      expect(i.queries).not.toContain("feedback_status_enum");
+    });
+
+    test("cannot create, update, or delete feedback_status_enum", () => {
+      expect(i).toHaveNoMutationsFor("feedback_status_enum");
+    });
+  });
+  describe("demoUser", () => {
```
Comment (DafyddLlyr) [inline hasura.planx.uk/tests/feedback_status.test.js:69]:
```suggestion
  });
  
  describe("demoUser", () => {
```

File: hasura.planx.uk/tests/flow_document_templates.test.js [hunk 6]
```diff
@@ -64,6 +64,20 @@ describe("flow_document_templates", () => {
       expect(i).toHaveNoMutationsFor("flow_document_templates");
     });
   });
+  describe("demoUser", () => {
```
Comment (DafyddLlyr) [inline hasura.planx.uk/tests/flow_document_templates.test.js:67]:
```suggestion
  });
  
  describe("demoUser", () => {
```

File: hasura.planx.uk/tests/flows.test.js [hunk 7]
```diff
@@ -140,6 +140,57 @@ describe("flows and operations", () => {
       expect(i.mutations).not.toContain("update_published_flows");
     });
   });
+  describe("demoUser", () => {
```
Comment (DafyddLlyr) [inline hasura.planx.uk/tests/flows.test.js:143]:
```suggestion
  });
  
  describe("demoUser", () => {
```

nit: Always nice to have a newline between these `describe()` blocks for readability.

**Review by DafyddLlyr — APPROVED** [review]
Looking good - a few small suggestions for tidy ups!

File: hasura.planx.uk/metadata/tables.yaml (hunk 2 above)
Comment (RODO94) [reply hasura.planx.uk/metadata/tables.yaml:569]: Found and resolved this! Thanks

**Pull request merged**